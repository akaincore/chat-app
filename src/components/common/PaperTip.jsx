import React from 'react';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
  paperMessage: {
    width: '50%',
    margin: '100px auto',
    padding: '50px',
    maxWidth: '400px',
    textAlign: 'center',
  },
});

const PaperTip = ({ classes, message }) => (
  <Paper className={classes.paperMessage}>{message}</Paper>
);

export default withStyles(styles)(PaperTip);
