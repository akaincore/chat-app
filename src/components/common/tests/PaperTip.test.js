/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PaperTip from '../PaperTip';

const mockProps = {
  message: 'foo bar baz',
};
describe('<PaperTip />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PaperTip {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<PaperTip {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
