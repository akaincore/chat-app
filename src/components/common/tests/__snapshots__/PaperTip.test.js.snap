// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaperTip /> renders correctly 1`] = `
<div
  className="MuiPaper-root-30 MuiPaper-elevation2-34 MuiPaper-rounded-31 PaperTip-paperMessage-29"
>
  foo bar baz
</div>
`;
